### ISSUE TITLE
Changing password breaks app

### ISSUE THREAD

**Issue by geez0x1** [IsmaelMartinez/teams-for-linux#461]
**Describe the bug**
Changing my account password breaks the Teams app, giving a message saying the tab needs to be refreshed. Doing so by pressing the button eventually brings it back to the same point.

**To Reproduce**
1. Start Teams succesfully
2. Change your account password
3. Close and start Teams again (cannot recall whether I was actively logged out)
4. See mentioned error

**Expected behavior**
Presented with initial log-in screen, as on first use.

**Desktop**
Ubuntu 21.04, latest snap package

**Workaround**
Deleting the contents of `~/snap/teams-for-linux/` solved the problem, but should not be necessary of course.

**Comment by jijojosephk** [comment]
@geez0x1 I've seen this in a different context. It has something to do with organisation setup. Ex: being idle for long. It says refresh but refreshing brings back to the same dialog again. I presume auth token expiration causing this. We've been working multi tenant/multi tab feature here #436 which would eventually solve this by allowing user to close the tab and open a fresh new session. @IsmaelMartinez have you experienced this? Do you think there is another solution?

**Comment by IsmaelMartinez** [comment]
Not seen that but it's probably some cached context. You can check in the know issues MD file. I think you might just be able to just remove some folders inside that folder. If you manage to find what ones, it might be possible to force a refresh of them.

**Comment by geez0x1** [comment]
A workaround is to delete `~/snap/teams-for-linux/current`; which does mean loggin in again and reconfiguring the settings (e.g. dark theme). This is obviously not ideal, but I was not able to find the particular cached context.

I noticed also that if I leave Teams open when putting my computer to stand-by at night, it is broken the next morning. This is probably a related issue but occurs relatively often. It might be related to my organisation requiring me to log in again every day (with 2FA).

**Comment by secdoc** [comment]
Same issue. Changed password today and unable to login in again. Get a 500 error.

Desktop - EndeavourOS Linux x86_64

**Comment by jijojosephk** [comment]
@secdoc could you explain where did you change the password. I think the same issue happens of teams is opened in web browser.

**Comment by secdoc** [comment]
> @secdoc could you explain where did you change the password. I think the same issue happens of teams is opened in web browser.

I changed my password within my works AD. Once changed I could not re-login to use the new password forcing me to have to uninstall and reinstall.

**Comment by IsmaelMartinez** [comment]
There is probably some caching going on (not of the password, but of a token) reason why deleting the config folder sorts the problem. Might be worth investigating that area.

**Comment by secdoc** [comment]
@IsmaelMartinez it is...when you log in the initial option is to keep logged in, but there is no way to change that after the password reset...so the only option is to not check the option in log on and to authenticate at each log on...

**Comment by jijojosephk** [comment]
@geez0x1 could you share the settings you would to preserve even after clearing the storage. In the upcoming release `1.0.41` there's a command line flag to clear the storage during the startup which is as same as deleting the config but would help some users who have less knowledge about doing such activities. 

I could find out that theme is stored 'Local Storage'.

**Comment by geez0x1** [comment]
@jijojosephk Sorry for the late reply. You mean the settings I would *like* to preserve I suppose? Personally the things I customise are typically dark theme and usage of British English. Although this problem probably would need to be addressed from the other direction; specifically clearing the relevant cached login context, keeping everything else.

**Comment by pebueno** [comment]
I was facing the same error, after changing my password for my e-mail on the browser I began having connection issues (like messages not being delivered) until a Teams window opened on the browser I didn't think it was my password change. I tried reinstalling Teams but after initializing it came straight to the error (a loop into trying to log in and then back to the error screen).
I'm not sure if I even typed my new password but suddenly I was logged in and the error was gone.
I'm using POP!_OS 22.04
